src/components/DataCollector/validateUserEventOptions.js

```diff
@@ -47,6 +47,9 @@ export default ({ options }) => {
     datasetId: string(),
     mergeId: string(),
     edgeConfigOverrides: validateConfigOverride,
+    advertising: objectOf({
+      handleAdvertisingData: string().default("disabled"),
```

Comment:
Please make this an `enumOf("disabled", "wait", "auto").default("disabled")` and make each of those values as constants that are shared across the component, like `src/constants/consentStatus.js`